react/leanShipping.js

```diff
@@ -343,47 +354,136 @@ function shouldShowFastest(cheapest, fastest) {
   const cheapestDetails = getStructuredOption(cheapest, CHEAPEST)
   const fastestDetails = getStructuredOption(fastest, FASTEST)
 
-  const hasCheapestLessPackages = cheapestDetails.packagesLength < fastestDetails.packagesLength
+  const hasCheapestLessPackages =
+    cheapestDetails.packagesLength < fastestDetails.packagesLength
   const fastestAndCheapestAreEqual = isEqual(cheapest, fastest)
 
   return fastest && !fastestAndCheapestAreEqual && !hasCheapestLessPackages
 }
 
-export function getLeanShippingOptions({
+function areSelectedSlasValid(logisticsInfo) {
+  const selectedSlas = logisticsInfo
+    .filter(logisticInfo => logisticInfo.selectedSla !== null)
+    .map(logisticInfo => logisticInfo.selectedSla)
+
+  return logisticsInfo.every(
+    logisticInfo =>
+      intersection(
+        logisticInfo.slas.map(sla => sla.id),
+        selectedSlas
+      ).length === 1
+  )
+}
+
+function setSelectedSlaIfValid({
+  logisticsInfo,
+  selectedSlas,
+  activeChannel,
+  isScheduledDeliveryActive,
+}) {
+  const newLogisticsInfo = setSelectedSla({
+    logisticsInfo,
+    selectedSlas,
+    activeChannel,
+    isScheduledDeliveryActive,
+  })
+  return areSelectedSlasValid(newLogisticsInfo) ? newLogisticsInfo : null
+}
+
+function getBestOption({
   logisticsInfo,
   activeChannel = DELIVERY,
   isScheduledDeliveryActive = false,
+  property,
 }) {
-  const arraysOfSlas = createArraysOfSlas(logisticsInfo)
-  const selectedSlas = {
-    cheapest: getMinSlaBy(arraysOfSlas, 'price'),
-    fastest: getMinSlaBy(arraysOfSlas, 'shippingEstimateInSeconds'),
+  const arrayOfSlas = createArrayOfSlas(logisticsInfo)
+
+  const currentSelectedSlas = arrayOfSlas.map(
+    (slas, index) =>
+      slas.filter(sla => sla.id === logisticsInfo[index].selectedSla)[0]
```

Comment:
again, not sure how large these arrays can be so this might not be a problem, but maybe you could iterate through it using a `for..of` and break on the first valid result instead of filtering the entire array

Reply:
Good point, I believe a nice middle ground would be using `find` instead of `filter`, looks good to you?
```suggestion
      slas.find(sla => sla.id === logisticsInfo[index].selectedSla)
```